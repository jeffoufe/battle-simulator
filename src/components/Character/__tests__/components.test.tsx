import React from 'react';
import { shallow } from 'enzyme';

import {
    CharacterContainer,
    CharacterInfoDice,
    CharacterInfo,
    CharacterImage,
    CharacterName,
    CharacterDice
} from '../components'


describe('Actions Components', () => {
    it('CharacterContainer', () => {
        const wrapper = shallow(
            <CharacterContainer order={0}>ActionsContainer</CharacterContainer>
        );
        expect(wrapper).toMatchSnapshot();
    })

    it('CharacterInfoDice', () => {
        const wrapper = shallow(
            <CharacterInfoDice order={0}>ActionsInfo</CharacterInfoDice>
        );
        expect(wrapper).toMatchSnapshot();
    })

    it('CharacterInfo', () => {
        const wrapper = shallow(
            <CharacterInfo order={0}>GameOver</CharacterInfo>
        );
        expect(wrapper).toMatchSnapshot();
    })

    it('CharacterImage', () => {
        const wrapper = shallow(
            <CharacterImage src='coucou'>GameOver</CharacterImage>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('img').props().src).toEqual('coucou');
    })

    describe('CharacterName', () => {
        const wrapper = shallow(
            <CharacterName order={0}>
                CharacterName
            </CharacterName>
        );

        expect(wrapper).toMatchSnapshot();
    })

    it('CharacterImage', () => {
        const wrapper = shallow(
            <CharacterDice order={0} />
        );
        expect(wrapper).toMatchSnapshot();
    })
})