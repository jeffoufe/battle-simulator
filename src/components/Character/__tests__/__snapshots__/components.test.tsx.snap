// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="sc-AxhUy hwArws"
  order={0}
>
  CharacterName
</div>
`;

exports[`Actions Components CharacterContainer 1`] = `
<div
  className="sc-AxjAm hrWJGs"
  order={0}
>
  ActionsContainer
</div>
`;

exports[`Actions Components CharacterImage 1`] = `
<img
  className="sc-AxhCb gSjfUP"
  src="coucou"
>
  GameOver
</img>
`;

exports[`Actions Components CharacterImage 2`] = `
<div
  className="sc-AxgMl deTFtb"
  order={0}
/>
`;

exports[`Actions Components CharacterInfo 1`] = `
<div
  className="sc-AxiKw hCvifY"
  order={0}
>
  GameOver
</div>
`;

exports[`Actions Components CharacterInfoDice 1`] = `
<div
  className="sc-AxirZ dLnpgW"
  order={0}
>
  ActionsInfo
</div>
`;
